Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -448,12 +472,71 @@ define([
          */
         hasAlphaChannel : {
             get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                if (!this.ready) {
+                    throw new DeveloperError('hasAlphaChannel must not be called before the imagery provider is ready.');
+                }
+                //>>includeEnd('debug');
                 return this._hasAlphaChannel;
             }
         }
     });
 
     /**
+     * Reinitializes this instance.  Reinitializing an instance already in use is supported, but it is not
+     * recommended because existing tiles provided by the imagery provider will not be updated.
+     *
+     * @param {Promise|Object} options Any of the options that may be passed to the {@see UrlTemplateImageryProvider} constructor.
+     */
+    UrlTemplateImageryProvider.prototype.reinitialize = function(options) {
+        var that = this;
+        return when(options).then(function(properties) {
+            //>>includeStart('debug', pragmas.debug);
+            if (!defined(properties)) {
+                throw new DeveloperError('properties is required.');
```

Comment:
This should probably say `options`, since that's what the user passed into `reinitialize` and the constructor.